code/game/objects/machinery/squad_manager.dm

```diff
@@ -0,0 +1,58 @@
+/obj/machinery/computer/squad_manager
+	name = "squad managment console"
+	desc = "A console for squad management. Allows squad leaders to manage their squad."
+	icon = 'icons/obj/machines/computer.dmi'
+	icon_state = "rdcomp"
+	interaction_flags = INTERACT_OBJ_UI
+
+/obj/machinery/computer/squad_manager/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "SquadManager", name)
+		ui.open()
+
+/obj/machinery/computer/squad_manager/ui_data(mob/user)
+	var/list/data = list()
+	data["active_squads"] = list()
+	for(var/datum/squad/squad AS in SSjob.active_squads[user.faction])
+		var/leader_name = squad.squad_leader?.real_name ? squad.squad_leader.real_name : "NONE"
+		data["active_squads"] += list(list("name" = squad.name, "leader" = leader_name, "color" = squad.color))
+	return data
+
+/obj/machinery/computer/squad_manager/ui_act(action, list/params)
+	. = ..()
+	if(.)
+		return
+	var/mob/living/carbon/human/user = usr
+	if(!ismarineleaderjob(user.job))
```

Comment:
I assume you're not enabling this in HvH, but this prevents SOM from using it even though you've made the code support it.
```suggestion
	if(!ismarineleaderjob(user.job) && !issommarineleaderjob(user.job))
```